extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const EXPERIMENT = 'amp-autocomplete';
+
+/** @const {string} */
+const TAG = 'amp-autocomplete';
+
+export class AmpAutocomplete extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.myText_ = 'hello world';
+
+    /** @private {?Element} */
+    this.container_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    userAssert(isExperimentOn(this.win, 'amp-autocomplete'),
+        `Experiment ${EXPERIMENT} is not turned on.`);
+    this.container_ = this.element.ownerDocument.createElement('div');
+    this.container_.textContent = this.myText_;
+    this.element.appendChild(this.container_);
+    this.applyFillContent(this.container_, /* replacedContent */ true);
+  }
+
+  /** @override */
+  layoutCallback() {
+    userAssert(isExperimentOn(this.win, 'amp-autocomplete'),
+        `Experiment ${EXPERIMENT} is not turned on.`);
+    // Actually load your resource or render more expensive resources.
+    return Promise.resolve();
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FIXED;
```

Comment:
I think we should start with `layout == Layout.CONTAINER`, since the `<input>` is going to be contained by the <amp-autocomplete>` tag, and the height of the element will be the height of the `<input>`.